src/rtt.cpp

```diff
@@ -0,0 +1,590 @@
+        Nan::ThrowError(message);
+        return;
+    }
+
+    log("===============================================\n");
+    log("Start of ");
+    log(baton->name);
+    log("\n");
+    log("===============================================\n");
+
+
+    baton->executeFunction = execute;
+    baton->returnFunction = ret;
+
+    uv_queue_work(uv_default_loop(), baton->req, ExecuteFunction, reinterpret_cast<uv_after_work_cb>(ReturnFunction));
+}
+
+void RTT::ExecuteFunction(uv_work_t *req)
+{
+    auto baton = static_cast<RTTBaton *>(req->data);
+
+    RTTErrorcodes_t executeError = baton->executeFunction(baton);
+
+    if (executeError != RTTSuccess)
+    {
+        baton->result = executeError;
+    }
+}
+
+void RTT::ReturnFunction(uv_work_t *req)
+{
+    Nan::HandleScope scope;
+
+    auto baton = static_cast<RTTBaton *>(req->data);
+    std::vector<v8::Local<v8::Value> > argv;
+
+    argv.push_back(ErrorMessage::getErrorMessage(baton->result, rtt_err_map, baton->name, logMessage, baton->lowlevelError));
+
+    if (baton->result != errorcode_t::JsSuccess)
+    {
+        for (uint32_t i = 0; i < baton->returnParameterCount; i++)
+        {
+            argv.push_back(Nan::Undefined());
+        }
+    }
+    else
+    {
+        if (baton->returnFunction != nullptr)
+        {
+            std::vector<v8::Local<v8::Value> > vector = baton->returnFunction(baton);
+
+            argv.insert(argv.end(), vector.begin(), vector.end());
+        }
+    }
+
+    baton->callback->Call(baton->returnParamterCount(), argv.data());
+
+    delete baton;
+}
+
+void RTT::resetLog()
+{
+    logMessage.clear();
+    appendToLog = true;
+    logItemCount = 0;
+}
+
+void RTT::log(std::string msg)
+{
+    if (logItemCount > 10000)
+    {
+        if (appendToLog)
+        {
+            logMessage = logMessage.append("The log has more than 10000 items. No more items will be logged.");
+            appendToLog = false;
+        }
+
+        return;
+    }
+
+    logMessage = logMessage.append(msg);
+    logItemCount++;
+}
+
+void RTT::logCallback(const char *msg)
+{
+    log(msg);
+}
+
+bool RTT::isStarted()
+{
+    if (!libraryLoaded) {
+        return false;
+    }
+
+    bool started;
+    dll_function.is_rtt_started(&started);
+
+    return started;
+}
+
+NAN_METHOD(RTT::Start)
+{
+    rtt_parse_parameters_function_t p = [&] (Nan::NAN_METHOD_ARGS_TYPE parameters, int &argumentCount) -> RTTBaton* {
+        auto baton = new RTTStartBaton();
+
+        baton->serialNumber = Convert::getNativeUint32(info[argumentCount]);
+        argumentCount++;
+
+        v8::Local<v8::Object> startOptions = Convert::getJsObject(parameters[argumentCount]);
+        StartOptions options(startOptions);
+        baton->hasControlBlockLocation = options.hasControlBlockLocation;
+        baton->controlBlockLocation = options.controlBlockLocation;
+        argumentCount++;
+
+        return baton;
+    };
+
+    rtt_execute_function_t e = [&] (RTTBaton *b) -> RTTErrorcodes_t {
```

Comment:
`rtt_execute_function_t e = [&] (RTTStartBaton *baton)`, so you don't need `static_cast` which should not even exist in C++.

Reply:
This suggested change results in a compile error, so will keep current implementation.

Reply:
I see I was wrong, since the function type requires RTTBaton*, however there are better casts probably for this case. Never mind.